src/rgw/rgw_rest_pubsub.cc

```diff
@@ -195,13 +195,17 @@ class RGWPSCreateTopicOp : public RGWOp {
       return 0;
     }
     if (ret == 0) {
-      if (result.user == s->owner.id ||
-          !s->cct->_conf->mandatory_topic_permissions) {
+      ret = verify_topic_owner_or_policy(
+          s, result, driver->get_zone()->get_zonegroup().get_name(),
+          rgw::IAM::snsGetTopicAttributes);
```

Comment:
shouldn't this use the `snsCreateTopic` action? it doesn't look like we've added that to rgw_iam_policy.h yet